system_tests/retry.py

```diff
@@ -0,0 +1,62 @@
+import time
+from functools import wraps
+
+import six
+
+
+class Retry(object):
+    """Retry class for retrying eventually consistent resources in testing."""
+
+    logger = None
+    exception = None
+    tries = None
+    delay = None
+    backoff = None
+
+    def __init__(self, exception, tries=4, delay=3, backoff=2, logger=None):
+        """Retry calling the decorated function using an exponential backoff.
+
+        :type exception: Exception or tuple of Exceptions
+        :param exception: The exception to check or may be a tuple of
+                          exceptions to check.
+
+        :type tries: int
+        :param tries: Number of times to try (not retry) before giving up.
+
+        :type delay: int
+        :param delay: Initial delay between retries in seconds.
+
+        :type backoff: int
+        :param backoff: Backoff multiplier e.g. value of 2 will double the
+                        delay each retry.
+
+        :type logger: logging.Logger instance
+        :param logger: Logger to use. If None, print.
+
+        :rtype: func
+        :returns: Retry wrapper function.
+        """
+
+        self.exception = exception
+        self.tries = tries
+        self.delay = delay
+        self.backoff = backoff
+        self.logger = logger.warning if self.logger else six.print_
+
+    def __call__(self, to_wrap):
+        @wraps(to_wrap)
+        def wrapped_function(*args, **kwargs):
+            while self.tries > 1:
```

Comment:
Using self here is problematic, since one could do the following:

```python
deco = Retry(MyExc, tries=2)

@deco
def f():
    # do stuff

f()

@deco
def g():
    # do stuff

g()
```

then `f` would consume all the retries before `g` does. You want `self` to be bound to whatever is being wrapped.

----

Also you want `self.tries > 0`